From: Justus Winter <justus@g10code.com>
Date: Mon, 23 Jan 2017 14:24:22 +0100
Subject: tools: Improve error handling.

* tools/gpgconf-comp.c (gp_component_change_options): Improve error
handling when reading from stdin.
--
Previously, errors encountered while reading the configuration changes
were ignored.

Signed-off-by: Justus Winter <justus@g10code.com>
(cherry picked from commit b0348fdb26637b0bcbd68a96c1746a1613b309af)
---
 tools/gpgconf-comp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/gpgconf-comp.c b/tools/gpgconf-comp.c
index 85eb80ab5..180fd65c2 100644
--- a/tools/gpgconf-comp.c
+++ b/tools/gpgconf-comp.c
@@ -3328,6 +3328,8 @@ gc_component_change_options (int component, estream_t in, estream_t out,
 
           change_one_value (option, runtime, flags, new_value, 0);
         }
+      if (length < 0 || gpgrt_ferror (in))
+	gc_error (1, errno, "error reading stream 'in'");
     }
 
   /* Now that we have collected and locally verified the changes,
